app/decorators/user_decorator.rb

```diff
@@ -1,9 +1,9 @@
 class UserDecorator < ApplicationDecorator
   delegate_all
 
-  def proposal_url(proposal)
+  def proposal_url(proposal, force_public_url=false)
     event = proposal.event
-    if model.staff_for? event
+    if model.staff_for?(event) && !force_public_url
```

Comment:
I'm wondering if we can't just flip this check too.  What if we see if the user is a speaker, if so give them the speaker link.  Otherwise send them to the reviewer's version.  If they are neither a speaker or staff, they reviewer's link will kick them out as not having staff access.